services/api-server/src/simcore_service_api_server/models/schemas/jobs.py

```diff
@@ -1,12 +1,83 @@
+import hashlib
 from datetime import datetime
 from enum import Enum
-from typing import Dict, List, Optional, Union
+from typing import Dict, List, Optional, Type, Union
 from uuid import UUID, uuid4
 
 from pydantic import BaseModel, Field, HttpUrl, conint, validator
 
 from ...models.schemas.files import File
-from ..api_resources import RelativeResourceName
+from ...models.schemas.solvers import Solver
+from ..api_resources import RelativeResourceName, compose_resource_name
+
+# JOB INPUTS/OUTPUTS ----------
+#
+#  - Wrappers for input/output values
+#  - Input/outputs are defined in service metadata
+#
+
+
+# FIXME: all ints and bools will be floats
+ArgumentType = Union[File, float, int, bool, str, None]
+KeywordArguments = Dict[str, ArgumentType]
+PositionalArguments = List[ArgumentType]
+
+
+class JobInputs(BaseModel):
+    # NOTE: this is different from the resource JobInput (TBD)
+    values: KeywordArguments
+
+    # TODO: gibt es platz fuer metadata?
+
+    class Config:
+        schema_extra = {
+            "example": {
+                "values": {
+                    "x": 4.33,
+                    "n": 55,
+                    "title": "Temperature",
+                    "enabled": True,
+                    "input_file": File(
+                        filename="input.txt", id="0a3b2c56-dbcd-4871-b93b-d454b7883f9f"
+                    ),
+                }
+            }
+        }
+
+    def compute_checksum(self):
+        # FIXME: this is wront since spaces can change checksum?
+        return hashlib.sha256(" ".join(self.json()).encode("utf-8")).hexdigest()
```

Comment:
I would pull out all the attributes from the object, sort them by name and they apply the hexdigest. Also consider this is a blocking operation.

Reply:
YEs. this is very wrong. I have done some changes combining the new hash function from pydantic but I am not 100% sure is totally correct. I have added some tests to keep an eye. For the moment, this feature is marked as experimental